k4GeneratorsConfig/src/xsection.cxx

```diff
@@ -62,20 +62,20 @@ bool k4GeneratorsConfig::xsection::processFile(){
 
   // open the edm4hep file
   m_reader->openFile(m_file);
-  
+
   // retrieve the RunInfo for the weight names, there should only be 1 entry per Run
   if ( m_reader->getEntries(podio::Category::Run) == 0 ){
     return false;
   }
   auto runinfo = podio::Frame(m_reader->readNextEntry(podio::Category::Run));
-  std::vector<std::string> weightNames = runinfo.getParameter<std::vector<std::string>>("WeightNames");
-  if ( weightNames.size() > 0 ){
-    std::cout << "k4GeneratorsConfig::Found Info on weight names: " << weightNames[0] << std::endl;
+  const auto& weightNames = runinfo.getParameter<std::vector<std::string>>("WeightNames");
```

Comment:
```suggestion
  const auto weightNames = runinfo.getParameter<std::vector<std::string>>("WeightNames");
```
You will get a value (the `std::optional`) here. Similar in other places. Lifetime extension might kick in, but I am not entirely sure.